distributed/comm/tests/test_comms.py

```diff
@@ -440,3 +640,130 @@ def test_tcp_many_listeners():
     check_many_listeners('tcp://127.0.0.1')
     check_many_listeners('tcp://0.0.0.0')
     check_many_listeners('tcp://')
+
+@gen_test()
+def test_inproc_many_listeners():
+    check_many_listeners('inproc://')
+
+
+#
+# Test deserialization
+#
+
+@gen.coroutine
+def check_listener_deserialize(addr, deserialize, in_value, check_out):
+    q = queues.Queue()
+
+    @gen.coroutine
+    def handle_comm(comm):
+        msg = yield comm.read()
+        q.put_nowait(msg)
+        yield comm.close()
+
+    with listen(addr, handle_comm, deserialize=deserialize) as listener:
+        comm = yield connect(listener.contact_address)
+
+    yield comm.write(in_value)
+    yield comm.close()
+
+    out_value = yield q.get()
+    check_out(out_value)
+
+@gen.coroutine
+def check_connector_deserialize(addr, deserialize, in_value, check_out):
+    q = queues.Queue()
+
+    @gen.coroutine
+    def handle_comm(comm):
+        msg = yield q.get()
+        yield comm.write(msg)
+        yield comm.close()
+
+    with listen(addr, handle_comm) as listener:
+        comm = yield connect(listener.contact_address, deserialize=deserialize)
+
+    q.put_nowait(in_value)
+    out_value = yield comm.read()
+    yield comm.close()
+    check_out(out_value)
+
+@gen.coroutine
+def check_deserialize(addr):
+    # Create a valid Serialized object
+    # (if using serialize(), it will lack a compression header)
```

Comment:
If no compression information is present in the header I would assume that the data is not compressed.  Is this not normally default behavior?